leaf-server/minecraft-patches/features/0251-Cache-block-path-type.patch

```diff
@@ -36,24 +36,30 @@ index 57aad048034005543a72556e990b53db8deebfee..6e6ee6f25f7cab0a6235d9e583ed5c01
 +    // Leaf end - Cache path type
  }
 diff --git a/net/minecraft/world/level/block/state/BlockBehaviour.java b/net/minecraft/world/level/block/state/BlockBehaviour.java
-index 6d522e9485fadd6fc0f350cb30ba5224aa046d4f..611887f5f8f218f5ec1ad19580f3123a60b20d46 100644
+index 6d522e9485fadd6fc0f350cb30ba5224aa046d4f..e2c049167f7b31051eddfbf5eda7b39389cecee0 100644
 --- a/net/minecraft/world/level/block/state/BlockBehaviour.java
 +++ b/net/minecraft/world/level/block/state/BlockBehaviour.java
-@@ -473,6 +473,7 @@ public abstract class BlockBehaviour implements FeatureElement {
+@@ -473,6 +473,8 @@ public abstract class BlockBehaviour implements FeatureElement {
          private boolean emptyCollisionShape;
          private boolean emptyConstantCollisionShape;
          private VoxelShape constantCollisionShape;
-+        public net.minecraft.world.level.pathfinder.PathType pathType; // Leaf - Cache path type
++        public byte pathType; // Leaf - Cache path type
++        public byte tagFlag; // Leaf - Cache path type
  
          private static void initCaches(final VoxelShape shape, final boolean neighbours) {
              ((ca.spottedleaf.moonrise.patches.collisions.shape.CollisionVoxelShape)shape).moonrise$isFullBlock();
-@@ -642,6 +643,12 @@ public abstract class BlockBehaviour implements FeatureElement {
+@@ -642,6 +644,17 @@ public abstract class BlockBehaviour implements FeatureElement {
              // Paper end - optimise collisions
          }
  
 +        // Leaf start - Cache path type
 +        public void initPathType() {
-+            pathType = net.minecraft.world.level.pathfinder.WalkNodeEvaluator.getPathTypeFromState(this.asState());
++            pathType = (byte) net.minecraft.world.level.pathfinder.WalkNodeEvaluator.getPathTypeFromState(this.asState()).ordinal();
++            tagFlag = this.asState().is(net.minecraft.tags.BlockTags.WALLS) ? (byte) 1 : (byte) 0;
++            tagFlag |= this.asState().is(net.minecraft.tags.BlockTags.FENCES) ? (byte) 2 : (byte) 0;
++            tagFlag |= this.asState().is(net.minecraft.tags.BlockTags.CLIMBABLE) ? (byte) 4 : (byte) 0;
++            tagFlag |= this.getBlock() instanceof net.minecraft.world.level.block.FenceGateBlock ? (byte) 8 : (byte) 0;
++            tagFlag |= this.getBlock() instanceof net.minecraft.world.level.block.TrapDoorBlock ? (byte) 16 : (byte) 0;
```

Comment:
Obviously faster to calculate the whole tagFlag in one go, and do it all using int, and then cast to byte only once in the end.